api/include/opentelemetry/nostd/mpark/variant.h

```diff
@@ -0,0 +1,1281 @@
+// MPark.Variant
```

Comment:
Not new, just moved here from `nostd/variant.h` to eventually eliminate - replace by `absl::variant`.